andela/mrm_api
# #165904360 Make structure_id unique

### What does this PR do?
Implements a check that ensures that an office structure cannot be added twice

### Description of the task to be completed?
- Generate migration file
- Update model definition
- Extend validate_structure_id function to check for structure_id uniqueness

### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/mrm_api.git`
2. Cd into the cloned repo and checkout to `ch-make-structure-id-unique-165904360` branch
3. Run the migration to setup the database (if you haven’t)  
5. Run the mutation below twice to trigger the structure_id uniqueness check
```
mutation {
  createOfficeStructure(data: [{structureId: "851ae8b3-48dd-46b5-89bc-ca3f8111ad87", name: "Office 1", level: 1, parentId: "1", tag: "office", position: 1, locationId: 2}, {structureId: "fd236595-71aa-407f-b417-f97ebadb94a0", name: "Office 2", level: 1, parentId: "1", tag: "office", position: 2, locationId: 2}, {structureId: "27fbc541-6c2f-437a-bb38-be9af64442c3", name: "Office 3", level: 1, parentId: "1", tag: "office", position: 2, locationId: 2}]) {
    structure {
      structureId
      name
      level
      parentId
      tag
      position
      locationId
    }
  }
}
```

### Any background context you want to provide?
Hitherto this implementation, the office_structure table does not impose the unique constraint on the stucture_id field and the above mutation doesn’t check for if the structure being created already exists in the database. This led to the possibility of having structures with the same structure ids which aren’t optimal. This PR exists to eliminate this anomaly. 

### Screenshot
* Check in action
<img width="1372" alt="Screenshot 2019-05-12 at 11 36 58 PM" src="https://user-images.githubusercontent.com/13919080/57588639-f6ac7500-750e-11e9-8fdd-5589ab66c121.png">

### What are the relevant pivotal tracker stories?
[#165904360](https://www.pivotaltracker.com/story/show/165904360)

### Checklist:
